### DISCUSSION TITLE
How do I obtain the path of the front-end plug-in installed at runtime

### DISCUSSION THREAD

**Discussion by zx616456607** [eclipse-theia/theia#13203] [General] [answered]

**Comment by xai** [comment, marked as answer]
Hi @zx616456607,

You can use the `pluginPath` attribute of a plugin to determine its deployment path.

As a practical example, let's create a Theia command that lists such information for all plugins:
```typescript
const showInstalledPlugins = {
    id: 'show-installed-plugins',
    label: "Show installed plugins"
};
context.subscriptions.push(theia.commands.registerCommand(showInstalledPlugins, (...args: any[]) => {
    let plugins = '';
    for (const plugin of theia.plugins.all){
        plugins += `${plugin.id} (${plugin.pluginType}) at ${plugin.pluginPath}\n`;
    }
    theia.window.showInformationMessage(`Installed plugins: ${plugins}`);
}));
```

Also, in practice, we recommend writing VSCode extensions or Theia extensions. Both are generally more effective and better supported compared to Theia plugins.

I hope this helps you to achieve what you need.

**Reply by zx616456607** [reply]
I found that using hostedPluginSupport, I can get the installed vscode plug-in information, including the installation path of the plug-in.

**Comment by JonasHelming** [comment]
@zx616456607 Is the answer sufficient? If so, could you mark it as answer above?

**Reply by zx616456607** [reply]
ok
